File: backend/Ecommerce.Domain/Entities/Product.cs
namespace Ecommerce.Domain.Entities;

public class Product
{
    public Guid Id { get; private set; } = Guid.NewGuid();
    public string Name { get; private set; }
    public decimal Price { get; private set; }
    public int Stock { get; private set; }
    
    public string ImageUrl { get; private set; }

    private Product() { }

    public Product(string name, decimal price, int stock, string imageUrl)
    {
        Name = name;
        Price = price;
        Stock = stock;
        ImageUrl = imageUrl;
    }
    
    public void UpdateImage(string imageUrl)
    {
        ImageUrl = imageUrl;
    }
}

File: frontend/src/types/product.ts
// export interface Product {
//   id: string;
//   name: string;
//   price: number;
//   stock: number;
//   imageUrl: string;
// }
//
// export interface PagedResult<T> {
//   items: T[];
//   page: number;
//   pageSize: number;
//   total: number;
// }

export type Product = {
  id: string;
  name: string;
  price: number;
  imageUrl: string;
  stock: number;
};